04_testing/outlier_checks.R

```diff
@@ -897,3 +898,365 @@ write_csv(
+    ),
+    legend.text = element_text(
+      size = 12,
+      family = "Roboto"
+    ),
+    legend.position = "bottom",
+    panel.grid.minor = element_blank(),
+    legend.background = element_rect(fill = "transparent"),
+    legend.box.background = element_rect(fill = "transparent"),
+    strip.text = element_text(
+      size = 16,
+      family = "Roboto"
+    ),
+    plot.margin = margin(10, 10, 10, 10)
+  )
+
+ggsave(
+  file.path(
+    file_paths$output_dir,
+    "graphs",
+    "Option 1",
+    "2022_percentage_of_outlier_frequency_by_pop.png"
+  ),
+  height = 6,
+  width = 8
+)
+
+write_csv(
+  df_outlier_pop,
+  file.path(
+    file_paths$output_dir,
+    "graphs",
+    "Option 1",
+    "datasets",
+    "2022_percentage_of_outlier_frequency_by_pop.csv"
+  )
+)
+
+## Simply plotting distribution of areas to show why using log
+
+df_pop_hist <- df_outliers %>%
+  mutate(
+    log_pop = log(affected_population),
+    log_pop_bins = cut(
+      log_pop,
+      breaks = 0:16,
+      labels = log_labs,
+      include.lowest = TRUE
+    )
+  ) %>%
+  group_by(
+    log_pop_bins
+  ) %>%
+  summarize(
+    n = n(),
+    .groups = "drop"
+  ) %>%
+  mutate(
+    perc_areas = n / sum(n)
+  )
+
+df_pop_hist %>%
+  ggplot() +
+  geom_bar(
+    aes(
+      x = fct_rev(log_pop_bins),
+      y = perc_areas
+    ),
+    stat = "identity",
+    fill = "#1EBFB3"
+  ) +
+  coord_flip() +
+  scale_y_continuous(
+    labels = scales::percent_format(
+      accuracy = 1
+    )
+  ) +
+  labs(
+    x = "Population size, disaggregated area",
+    y = "% of areas",
+    title = "% of units of analysis by population size"
+  ) +
+  theme_minimal() +
+  theme(
+    plot.title = element_text(
+      face = "bold",
+      size = 22,
+      margin = margin(10, 10, 10, 10, "pt"),
+      family = "Roboto"
+    ),
+    plot.background = element_rect(
+      fill = "white"
+    ),
+    axis.text = element_text(
+      face = "bold",
+      size = 10,
+      family = "Roboto"
+    ),
+    legend.text = element_text(
+      size = 12,
+      family = "Roboto"
+    ),
+    legend.position = "bottom",
+    panel.grid.minor = element_blank(),
+    legend.background = element_rect(fill = "transparent"),
+    legend.box.background = element_rect(fill = "transparent"),
+    strip.text = element_text(
+      size = 16,
+      family = "Roboto"
+    )
+  )
+
+ggsave(
+  file.path(
+    file_paths$output_dir,
+    "graphs",
+    "Option 1",
+    "2022_population_histogram.png"
+  ),
+  height = 6,
+  width = 8
+)
+
+write_csv(
+  df_pop_hist,
+  file.path(
+    file_paths$output_dir,
+    "graphs",
+    "Option 1",
+    "datasets",
+    "2022_population_histogram.csv"
+  )
+)
+
+## Look at number of units of analysis and outliers
+
+df_outliers_num <- df_outliers %>%
+  group_by(
+    adm0_pcode
+  ) %>%
+  summarize(
+    units = n(),
+    num_sectors = length(unique(sector)),
+    pct_outliers = sum(is_any_outlier) / units,
+    .groups = "drop"
+  )
+
+df_outliers_num %>%
```

Comment:
showing the countries as legend would probably explain the crazy figures from the outlier graphs.

Reply:
Hmm, do we think it's only country level phenomena and there's no general trend? What do you think?